Test/Tasty/Golden/Internal.hs

```diff
@@ -62,18 +65,55 @@ instance IsOption SizeCutoff where
   optionHelp = return "hide golden test output if it's larger than n bytes"
   optionCLParser = mkOptionCLParser $ metavar "n"
 
+-- | When / whether to delete the test output file,
+-- when there is a golden file
+data DeleteOutputFile
+  = Always -- ^ Always delete the output file
+  | OnPass -- ^ Delete the output file if the test passes
+  | OnFail -- ^ Ditto, if the test fails
+  | Never  -- ^ Never delete the output file (default)
+  deriving (Eq, Ord, Typeable, Show)
```

Comment:
Can we imagine a scenario under which `OnFail` could be helpful? If not, I suggest dropping that option, as it would just unnecessarily confuse the user.

Reply:
I sometimes want to look only at the trace output on the screen and not trigger a subsequent phase of the test pipeline.  But I agree that it's not a typical use case.  I've relegated it to the last option and added a comment to bring that out.

Reply:
`Always` is fine; it was `OnFail` that I was wondering about.

Reply:
Similar reasoning.  It seems an odd thing to do as part of a normal regression testing run, I agree, but I can see it being convenient to do, at the flick of a switch, when debugging, eg, to avoid the results being taken up by a subsequent step in a testing pipeline.